### PR TITLE
Added SimilarityTransform and slightly refactored existing transforms.

### PR DESCRIPTION
**Pull request by marcelluethi**
The refactoring mitigates minor annoyances, such as for example
- Incorrect naming of ProductTransforms (should be CompositeTransform)
- Missing type annotation for inverses (e.g. in RigidTransformation)
- Center of rotation as default arguments
- Some Missing factory methods for creating transformations

### CODE REVIEW

File: src/main/scala/scalismo/registration/TransformationSpace.scala [hunk 1]
```diff
@@ -618,34 +654,100 @@ object RigidTransformationSpace {
...
+private class SimilarityTransformationScalingFollowingRigid[D <: Dim: NDSpace](val scaling: ScalingTransformation[D],
+  val rigidTransformation: RigidTransformation[D])
+    extends CompositeTransformation[D](scaling, rigidTransformation) with SimilarityTransformation[D] {
 
+  override def inverse: SimilarityTransformation[D] = new SimilarityTransformationRigidFollowingScaling[D](rigidTransformation.inverse, scaling.inverse)
+}
+
+private class SimilarityTransformationRigidFollowingScaling[D <: Dim: NDSpace](val rigidTransformation: RigidTransformation[D],
+  val scaling: ScalingTransformation[D])
+    extends CompositeTransformation[D](scaling, rigidTransformation) with SimilarityTransformation[D] {
```
Comment (Ghazi-Bouabene) [inline src/main/scala/scalismo/registration/TransformationSpace.scala]: Shouldn't the order be different here: i.e. (rigidTransformation, scaling)

File: src/main/scala/scalismo/registration/TransformationSpace.scala [hunk 2]
```diff
@@ -618,34 +654,100 @@ object RigidTransformationSpace {
...
+ * @constructor Returns a parametric space generating anisotropic similarity transforms
+ * @param center : center of rotation used in the rigid transform
+ */
+case class SimilarityTransformationSpace[D <: Dim: NDSpace: CreateRotationSpace: ScalingSpace.Create](center: Point[D])
+    extends ProductTransformationSpace[D, ScalingTransformation[D], RigidTransformation[D]](ScalingSpace[D](), RigidTransformationSpace[D](center)) {
+
+  /**
+   * Returns a D-dimensional c similarity transform that performs the rigid transofrmation first, then a scaling.
+   *
+   * @param p parameter vector for the transform. This must be a concatenation of the rigid transform parameters first, then scaling parameters
```
Comment (Ghazi-Bouabene) [inline src/main/scala/scalismo/registration/TransformationSpace.scala]: The API doc description does not match the code below. When using splitProductParameterVector, you assume that the scaling parameters come first

File: src/test/scala/scalismo/registration/TransformationTests.scala [hunk 3]
```diff
@@ -249,4 +249,31 @@ class TransformationTests extends ScalismoTestSuite {
...
+    }
+
+  }
+
+  describe("A 3D similarity transform") {
+
+    val rigidTransformation = RigidTransformation(TranslationTransform(Vector3D(2.0, 5.0, 0.0)), RotationTransform(Math.PI / 2.0, 0.0, 0.0, Point3D(0, 0, 0)))
+    val similarityTransform = SimilarityTransformation(ScalingTransformation[_3D](2.0), rigidTransformation)
+
+    it("correctly transforms a 3D point") {
```
Comment (Ghazi-Bouabene) [inline src/test/scala/scalismo/registration/TransformationTests.scala]: not sure if this is really a 3D test :p

Comment (marcelluethi) [inline src/test/scala/scalismo/registration/TransformationTests.scala]: Agreed :-)

File: src/main/scala/scalismo/registration/TransformationSpace.scala (hunk 1 above)
Comment (marcelluethi) [inline src/main/scala/scalismo/registration/TransformationSpace.scala]: nice catch

**Pull request merged**